src/PropertyAnnotators/ApprovedByPropertyAnnotator.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+namespace SESP\PropertyAnnotators;
+
+use SESP\AppFactory;
+use SESP\PropertyAnnotator;
+use SESP\DatabaseLogReader;
+use SMW\DIWikiPage;
+use SMWDataItem as DataItem;
+use SMW\DIProperty;
+use SMW\SemanticData;
+use Title;
+use User;
+
+/**
+ * @private
+ * @ingroup SESP
+ *
+ * @license GNU GPL v2+
+ */
+class ApprovedByPropertyAnnotator implements PropertyAnnotator {
+
+	/**
+	 * Predefined property ID
+	 */
+	const PROP_ID = '___APPROVEDBY';
+
+	/**
+	 * @var AppFactory
+	 */
+	private $appFactory;
+
+	/**
+	 * @var Integer|null
+	 */
+	private $approvedBy;
+
+	/**
+	 * @param AppFactory $appFactory
+	 */
+	public function __construct( AppFactory $appFactory ) {
+		$this->appFactory = $appFactory;
+	}
+
+	/**
+	 * @since 2.0
+	 *
+	 * @param User $approvedBy
+	 */
+	public function setApprovedBy( $approvedBy ) {
+		$this->approvedBy = $approvedBy;
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public function isAnnotatorFor( DIProperty $property ) {
+		return $property->getKey() === self::PROP_ID;
+	}
+
+	public function getDataItem() {
+		if ( $this->approvedBy instanceof User ) {
+			$userPage = $this->approvedBy->getUserPage();
+
+			if ( $userPage instanceof Title ) {
+				return DIWikiPage::newFromTitle( $userPage );
+			}
+		}
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public function addAnnotation(
+		DIProperty $property, SemanticData $semanticData
+	) {
+		if ( $this->approvedBy === null && class_exists( 'ApprovedRevs' ) ) {
+			$logReader = $this->appFactory->newDatabaseLogReader(
+				$semanticData->getSubject()->getTitle(), 'approval'
```

Comment:
SMW normally doesn't validate whether a subject is correct in terms of the MediaWiki title validation, so when you call `$semanticData->getSubject()->getTitle()` it internally calls `Title::makeTitleSafe` which by definition can return a `NULL` [0] but since you assume to always hold a strict `Title` in the constructor, what happens when you get a `NULL` from `Title::makeTitleSafe`?

[0] https://phabricator.wikimedia.org/search/query/SUr6RqZWQ7fb/#R